wled00/wled_server.cpp

```diff
@@ -228,70 +235,67 @@ void initServer()
     if (handleFileRead(request, "/simple.htm")) return;
     if (handleIfNoneMatchCacheHeader(request)) return;
     AsyncWebServerResponse *response = request->beginResponse_P(200, "text/html", PAGE_simple, PAGE_simple_L);
-    response->addHeader(F("Content-Encoding"),"gzip");
+    response->addHeader(FPSTR(s_content_enc),"gzip");
     setStaticContentCacheHeaders(response);
     request->send(response);
   });
 #endif
 
   server.on("/iro.js", HTTP_GET, [](AsyncWebServerRequest *request){
     AsyncWebServerResponse *response = request->beginResponse_P(200, "application/javascript", iroJs, iroJs_length);
-    response->addHeader(F("Content-Encoding"),"gzip");
+    response->addHeader(FPSTR(s_content_enc),"gzip");
     setStaticContentCacheHeaders(response);
     request->send(response);
   });
   
   server.on("/rangetouch.js", HTTP_GET, [](AsyncWebServerRequest *request){
     AsyncWebServerResponse *response = request->beginResponse_P(200, "application/javascript", rangetouchJs, rangetouchJs_length);
-    response->addHeader(F("Content-Encoding"),"gzip");
+    response->addHeader(FPSTR(s_content_enc),"gzip");
     setStaticContentCacheHeaders(response);
     request->send(response);
   });
   
-  createEditHandler(correctPIN);
+  createEditHandler(correctPIN && !otaLock);
```

Comment:
OTA lock should have no influence on ability to visit /edit page.
Security setting cannot be changed using /edit.

Reply:
`/edit` page is able to set (not retrieve sec) arbitrary settings, but more critically, able to alter the pages displayed to the browser, which is a security risk. It can even be seen as a form of OTA update :)
PIN is an adequate protection, but if the user has OTA lock active before and after the update, they would still expect `/edit` to be locked. I will change the label of OTA to make clear that it also locks edit though.

Reply:
Sorry but I fail to understand how a user can update if OTA is locked? If you mean serially then all measures are irrelevant since with physical access anything is possible.
If OTA is locked it will prevent any kind of updates (not true ATM due to bug) to firmware even if PIN is entered and that should be the main purpose of it IMO (which also holds for WiFi settings).

Reply:
They can't update the firmware, but they _can_ arbitrarily change the `index.htm` served to clients.

Regarding ArduinoOTA: I think it should stay separate from OTA lock - most common use case for OTA lock is allowing access to port 80 only. This allows local updates via ArduinoOTA without having to unlock OTA each time. If ArduinoOTA is disabled via the checkbox and OTA locked, it can't be re-enabled without the correct OTA password. Should add a note though that ArduinoOTA will remain active unless disabled manually.

Reply:
I still think that locking OTA should be consistent for both options: web update and Arduino OTA. Otherwise an adversary can update firmware even if a user thinks he/she locked OTA updates and considers WLED secure.

> `/edit` page is able to set (not retrieve sec) arbitrary settings, but more critically, able to alter the pages displayed to the browser

Locking settings (and `/edit`) using PIN code should be sufficient for most cases where WLED can be put to public use (with the warning about non encrypted transfer). Removing fake `index.htm` can be done without too much effort in case when adversary intercepted PIN and uploaded fake `index.htm`. Combined with OTA lock it can prevent most malicious attacks (except carefully planned ones) even with plain HTTP. I would not expect `/edit` to be locked when OTA is disabled/locked. It has become a feature that was suggested as a help in case of many problem solving options. It will also prevent uploading _ledmaps_ and such which should not be a condition of OTA lock.

PIN in itself will also prevent updating OTA and will lend a grace period of 15 minutes (when entered) in which time you can do OTA (Arduino or web) if you need to update.